### DISCUSSION TITLE
Lifespan causes deprecated startup/shutdowns to silently fail

### DISCUSSION THREAD

**Discussion by JustinDula** [fastapi/fastapi#9604] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

from contextlib import asynccontextmanager

@asynccontextmanager
async def lifespan(app: FastAPI):
    # Load the ML model
    print("lifespan start")
    yield
    # Clean up the ML models and release the resources
    print("lifespan end")


USE_LIFESPAN = True

app = FastAPI(lifespan=lifespan if USE_LIFESPAN else None)

@app.on_event("startup")
def startup():
	print("startup event")
        
@app.on_event("shutdown")
def startup():
	print("startup event")
```

### Description

Run the application with `USE_LIFESPAN = False`, and `startup()` and `shutdown()` are called normally, as expected.

However, when using the lifespan function (`USE_LIFESPAN = True`), the `startup()` and `shutdown()` function are not called.  Instead, only the `lifespan()` function is called. 

While this behavior may be intentional, it is not clearly indicated in the [documentation ](https://fastapi.tiangolo.com/advanced/events/#lifespan). I uncovered this behavior while debugging a library that started silently failing when I introduced a lifespan to my application, as it was using the events under the hood. I believe it would be appropriate to raise a warning or exception if `lifespan` is set but there are `startup` or `shutdown` events. Additionally, in order to enable the use of existing libraries that may rely on these events, it may be best to retain support for startup and shutdown and add an option to allow for the two to coexist when creating the FastAPI application. 

### Operating System

Windows

### FastAPI Version

0.95.2

### Python Version

3.11.3

**Comment by Kludex** [comment, marked as answer]
This is entirely on Starlette's side, not FastAPI. Please open a discussion over there.

**Reply by JustinDula** [reply]
Ah, okay. Might still be good to put a note in the FastAPI docs page for lifespan events that the two are mutually exclusive.

**Reply by Kludex** [reply]
The best thing to do is to not have it silent. Starlette should solve this.

**Reply by stratosgear** [reply]
I was just pulling my hair for the last hour or so, why my startup/shutdown events suddenly were not working. (well, suddenly: since I introduced a call to lifespan, that I failed to immediately notice the startup/shutdown missing calls).

Silently ignoring them, and NOT raising any warning or errors is REALLY tricky to debug.... :(

**Comment by ewianda** [comment]
Also just found out that setting `lifespan=off` also causes this issue
